distributed/shuffle/tests/test_shuffle.py

```diff
@@ -1010,6 +1010,41 @@ async def test_closed_worker_between_repeats(c, s, w1, w2, w3):
     await clean_scheduler(s)
 
 
+@pytest.mark.slow
+@gen_cluster(client=True, nthreads=[("", 1)], Worker=Nanny)
+async def test_restart_cluster_between_repeats(c, s, a):
+    scheduler_extension = s.extensions["shuffle"]
+    df = dask.datasets.timeseries(
+        start="2000-01-01",
+        end="2000-01-10",
+        dtypes={"x": float, "y": float},
+        freq="100 s",
+        seed=42,
+    )
+
+    await c.compute(dd.shuffle.shuffle(df, "y", shuffle="p2p"))
+    await clean_scheduler(s)
+
+    assert scheduler_extension.tombstones
+
+    # Cannot rerun forgotten shuffle due to tombstone
+    with pytest.raises(RuntimeError, match="shuffle_transfer"):
+        await c.compute(dd.shuffle.shuffle(df, "y", shuffle="p2p"))
```

Comment:
Hold on, this is concerning. We cannot rerun the same shuffle even after it finished successfully?

Reply:
See also https://github.com/dask/distributed/blob/401b51dd1363f30b163c9811efd7651379a32772/distributed/shuffle/tests/test_shuffle.py#L892

Reply:
xref https://github.com/dask/distributed/issues/7452